fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/query/plan/plans/ExplodePlanTest.java

```diff
@@ -0,0 +1,115 @@
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+public class ExplodePlanTest {
+
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    private static final class ExplodeCursorBuilder {
+
+        @Nonnull
+        private final RecordQueryPlan explodePlan;
+
+        @Nonnull
+        private Optional<Integer> skip;
+
+        @Nonnull
+        private Optional<Integer> limit;
+
+        private ExplodeCursorBuilder() {
+            explodePlan = generateExplodePlan();
+            skip = Optional.empty();
+            limit = Optional.empty();
+        }
+
+        @Nonnull
+        ExplodeCursorBuilder withSkip(int skip) {
+            this.skip = Optional.of(skip);
+            return this;
+        }
+
+        @Nonnull
+        ExplodeCursorBuilder withLimit(int limit) {
+            this.limit = Optional.of(limit);
+            return this;
+        }
+
+        @Nonnull
+        @SuppressWarnings("DataFlowIssue") // explode transposes the underlying constant array Value, it does not strictly require a record store instance.
+        RecordCursor<QueryResult> build() {
+            final var executionPropertiesBuilder = ExecuteProperties.newBuilder();
+            skip.ifPresent(executionPropertiesBuilder::setSkip);
+            limit.ifPresent(executionPropertiesBuilder::setReturnedRowLimit);
+            final var executionProperties = executionPropertiesBuilder.build();
+            return explodePlan.executePlan(null, EvaluationContext.EMPTY, null, executionProperties);
+        }
+
+        @Nonnull
+        private static RecordQueryPlan generateExplodePlan() {
+            final Value collectionValue = LiteralValue.ofList(List.of(1, 2, 3, 4, 5, 6, 7, 8, 9, 10));
+            return new RecordQueryExplodePlan(collectionValue);
+        }
+
+        @Nonnull
+        public static ExplodeCursorBuilder instance() {
+            return new ExplodeCursorBuilder();
+        }
+    }
+
+    private static void verifyCursor(@Nonnull final RecordCursor<QueryResult> actualCursor,
+                                     @Nonnull final List<Integer> expectedResults,
+                                     boolean verifyLimitExceeded) {
+        for (final var expectedValue : expectedResults) {
+            final var result = actualCursor.getNext();
+            Assertions.assertTrue(result.hasNext());
+            Assertions.assertNotNull(result.get());
+            Assertions.assertEquals(expectedValue, Objects.requireNonNull(result.get()).getDatum());
+        }
+        if (verifyLimitExceeded) {
+            final var result = actualCursor.getNext();
+            Assertions.assertFalse(result.hasNext());
+            Assertions.assertEquals(RecordCursor.NoNextReason.RETURN_LIMIT_REACHED, result.getNoNextReason());
+        }
+    }
+
+    @Test
+    void explodeWithSkipAndLimitWorks() {
+        verifyCursor(ExplodeCursorBuilder.instance().withLimit(1).build(), ImmutableList.of(1), true);
```

Comment:
Relatively minor: it would be nice if this were structured as a `@ParameterizedTest` with the different cases as arguments rather than one test with all of the test cases. That way, if one of them fails, we know which one(s). Otherwise, a failure in an earlier case can mask failures in other test cases.

Reply:
Done, I also tried to make the parameterized test output look more readable:

<img width="783" alt="image" src="https://github.com/user-attachments/assets/7f0d5f61-0d36-4697-8d41-377c0cc66782" />